docs/docs-beta/docs/tutorial/pinecone.md

```diff
@@ -0,0 +1,311 @@
+        create table if not exists reviews as (
+          select *
+          from read_json(
+            '{url}',
+            ignore_errors = true
+          )
+        );
+    """
+    with duckdb_resource.get_connection() as conn:
+        conn.execute(query)
+```
+
+With our DuckDB tables created, we can now query them like any other SQL table. Our third asset will join and filter the data and then return a DataFrame (we will also `LIMIT` the results to 500):
+```python
+# asssets.py
+@dg.asset(
+    kinds={"duckdb"},
+    group_name="ingestion",
+)
+def enriched_reviews(
+    duckdb_resource: dg_duckdb.DuckDBResource,
+    graphic_novels,
+    reviews,
+) -> pd.DataFrame:
+    query = f"""
+        select
+          reviews.review_id as review_id,
+          graphic_novels.title as title,
+          reviews.rating as rating,
+          reviews.review_text as text
+        from graphic_novels
+        left join reviews
+          on graphic_novels.book_id = reviews.book_id
+        where graphic_novels.language_code = 'eng'
+        limit 500
+    """
+    with duckdb_resource.get_connection() as conn:
+        return conn.execute(query).fetch_df()
+```
+
+Now that the data has been prepared, we are ready to work with our vector database.
+
+### Vector Database
+We can begin by creating the index within our vector database. A vector database is a specialized database designed to store, manage, and retrieve high-dimensional vector embeddings, enabling efficient similarity search and machine learning tasks. There are many different vector databases available. For this demo we will use Pinecone which is a cloud based vector database that offers a [free tier](https://app.pinecone.io/) that can help us get started.
+
+:::note
+After you have your API key from Pinecone. You will want to set it to the environment variable `PINECONE_API_KEY`:
+```
+export PINECONE_API_KEY={YOUR PINECONE API KEY}
+```
+:::
+
+Now that we have a Pinecone account we can incorporate Pinecone into our code. To help make everything consistent, we will create a Pinecone resource similar to the Dagster maintained DuckDB resource we used to ingest the Goodreads data. Creating a custom resource is fairly simple, the only information we need to authenticate with the Pinecone client is their API key: 
+```python
+#resources.py
+class PineconeResource(ConfigurableResource):
+    api_key: str = Field(
+        description=(
+            "Pinecone API key. See https://docs.pinecone.io/reference/api/introduction"
+        )
+    )
+
+    _client: Pinecone = PrivateAttr()
+
+    def setup_for_execution(self, context: InitResourceContext) -> None:
+        self._client = Pinecone(
+            api_key=self.api_key,
+        )
+    ...
+```
+
+For our methods we will want the ability to create an index, retrieve a index so we can upsert records, and the ability to embed inputs. You can see all the details in the repo but this is what the `create_index` method looks like:
```

Comment:
This might be a silly suggestion, but can we explain what an index is in the context of vector databases?